packages/storage/src/service.ts

```diff
@@ -16,36 +16,129 @@
  */
 
 import { FirebaseApp } from '@firebase/app-types';
-import * as args from './implementation/args';
 import { Location } from './implementation/location';
 import { FailRequest } from './implementation/failrequest';
 import { Request, makeRequest } from './implementation/request';
 import { RequestInfo } from './implementation/requestinfo';
 import { XhrIoPool } from './implementation/xhriopool';
-import { Reference } from './reference';
+import { Reference, getParent, getChild } from './reference';
 import { Provider } from '@firebase/component';
 import { FirebaseAuthInternalName } from '@firebase/auth-interop-types';
 import { FirebaseOptions } from '@firebase/app-types-exp';
 import * as constants from '../src/implementation/constants';
 import * as errorsExports from './implementation/error';
 
+export function isUrl(path?: string): boolean {
+  return /^[A-Za-z]+:\/\//.test(path as string);
+}
+
+/**
+ * Returns a firebaseStorage.Reference for the given url.
+ */
+function refFromURL(service: StorageService, url: string): Reference {
+  return new Reference(service, url);
+}
+
+/**
+ * Returns a firebaseStorage.Reference for the given path in the default
+ * bucket.
+ */
+function refFromPath(
+  ref: StorageService | Reference,
+  path?: string
+): Reference | null {
+  if (ref instanceof StorageService) {
+    const service = ref;
+    if (service.bucket_ == null) {
+      throw errorsExports.noDefaultBucket();
+    }
+    const reference = new Reference(service, service.bucket_!);
+    if (path != null) {
+      return refFromPath(service, path);
+    } else {
+      return reference;
+    }
+  } else {
+    // ref is a Reference
+    if (path === '..') {
+      return getParent(ref);
+    } else if (typeof path === 'string') {
+      return getChild(ref, path);
+    } else {
+      return ref;
+    }
+  }
+}
 /**
- * A service that provides firebaseStorage.Reference instances.
+ * Returns a firebaseStorage.Reference for the given path in the default
+ * bucket.
+ */
+/**
+ * Returns a storage Reference for the given url, or given path in the
+ * default bucket.
+ * @param serviceOrRef `Storage` instance or storage `Reference`.
+ * @param pathOrUrl Storage path, or URL. If empty, returns root reference (if Storage
+ * instance provided) or returns same reference (if Reference provided).
+ */
+export function ref(
+  serviceOrRef: StorageService | Reference,
+  pathOrUrl?: string
+): Reference | null {
+  if (pathOrUrl && isUrl(pathOrUrl)) {
+    if (serviceOrRef instanceof StorageService) {
+      return refFromURL(serviceOrRef, pathOrUrl);
+    } else {
+      throw errorsExports.invalidArgument(
```

Comment:
Does this namespace import break tree-shaking?

Reply:
Switched to named imports.